client/document.go

```diff
@@ -274,14 +274,14 @@ func validateFieldSchema(val any, field FieldDefinition) (NormalValue, error) {
 		return NewNormalString(v), nil
 
 	case FieldKind_STRING_ARRAY:
-		v, err := getArray(val, getString)
+		v, err := getArray(val, getString, field.Size)
 		if err != nil {
 			return nil, err
 		}
 		return NewNormalStringArray(v), nil
 
 	case FieldKind_NILLABLE_STRING_ARRAY:
-		v, err := getNillableArray(val, getString)
+		v, err := getNillableArray(val, getString, field.Size)
```

Comment:
thought: out of scope, can't believe these cases are untested, specially in a validation function, and all other cases in this function.